### ISSUE TITLE
Merge dicts using union operator producing invalid syntax for dictionary comprehensions

### ISSUE THREAD

**Issue by RobbieClarken** [asottile/pyupgrade#504]
Running `pyupgrade --py39-plus` on the following code is producing invalid syntax:

**In:**

```python
A = {"w": "x"}
B = {"y": "z"}
C = {
    **{v: k for k, v in A.items()},
    **B,
}
```

**Out:**

```python
A = {"w": "x"}
B = {"y": "z"}
C = (
    {v: k for k | v in A.items()},
    B
)
```

I believe it should be rewritten as:

```python
A = {"w": "x"}
B = {"y": "z"}
C = (
    {v: k for k, v in A.items()} |
    B
)
```

This appears to be due to the rule introduced in https://github.com/asottile/pyupgrade/pull/436 and impacts pyupgrade v2.16.0 through to the current version (v2.23.0).

**Comment by MarcoGorelli** [comment]
Indeed, thanks for the report @RobbieClarken !

I think the fix is to search backwards for commas, rather than forwards

**Comment by RobbieClarken** [comment]
Thanks @MarcoGorelli - the fix added in #505 worked as intended on my codebase. 👏
